File: ko2keggc.rb
#!/usr/bin/env ruby

strKO = nil
hashKEGGs = {}
STDIN.each do |strLine|
	if( strLine =~ /^ENTRY\s+(K\d+)/ )
		strKO = $1
	elsif( strLine =~ /(?:(?:PATH)|(?:BR)):(ko\d+)/ )
#	elsif( strLine =~ /(?:PATH):(ko\d+)/ )
		if( !( astrKOs = hashKEGGs[$1] ) )
			hashKEGGs[$1] = astrKOs = []; end
		astrKOs.push( strKO ); end; end

puts( hashKEGGs.map do |strKEGG, astrKOs|
	strKEGG + "\t" + astrKOs.join( "\t" ); end.join( "\n" ) )

File: ko2keggc.py
#!/usr/bin/env python

import re
import sys

strKO = None
hashKEGGs = {}
for strLine in sys.stdin:
	mtch = re.search( '^ENTRY\s+(K\d+)', strLine )
	if mtch:
		strKO = mtch.group( 1 )
		continue
#	mtch = re.search( '(?:(?:PATH)|(?:BR)):(ko\d+)', strLine )
	mtch = re.search( 'PATH:(ko\d+)', strLine )
	if mtch:
		hashKEGGs.setdefault( mtch.group( 1 ), set() ).add( strKO )

for strKEGG, setKOs in hashKEGGs.items( ):
	print( "\t".join( [strKEGG] + list(setKOs) ) )
